smoker/client/__init__.py

```diff
@@ -290,15 +292,18 @@ class Host(object):
 
     _result  = None
 
-    def __init__(self, address):
+    def __init__(self, address, config):
```

Comment:
you probably don't need to pass the whole config here, you could use an advantage of default parameter value like 
```
def __init__(self, address, port=8086)
```
and move the logic of checking whether port was supplied in the config to some calling class/function